modules/styling/lib/cs.ts

```diff
@@ -239,12 +248,12 @@ type ToString<T> = string & T;
  * ```
  */
 export type DefaultedVarsMapToCSSVarNames<
-  V extends Record<string, string> | Record<string, Record<string, string>>,
+  V extends DefaultedVarsShape,
   ID extends string | never
 > = [ID] extends [never]
   ? V extends Record<string, string>
-    ? Record<keyof V, string>
-    : {[K in keyof V]: Record<keyof V[K], string>}
+    ? {[K in keyof V]: V[K]}
+    : {[K in keyof V]: {[K2 in keyof V[K]]: V[K][K2]}}
```

Comment:
This change fixes JSDoc for variables:

```tsx
const myStencil = createStencil({
  vars: {
    /** The JSDoc comment */
    color: 'red'
  },
  base({color}) {
    // color would not have JSDoc
  }
})

// before
myStencil.vars.color // would not have JSDoc
```

Now:

![image](https://github.com/Workday/canvas-kit/assets/338257/98eb4141-a423-48a8-80fb-bbe03eebbed0)

![image](https://github.com/Workday/canvas-kit/assets/338257/4d00ec17-c998-438a-9c8b-d527c6873fd8)